app/controllers/mainController.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable consistent-return */
+// Utilisation de PSQL
+
+const dataMapper = require('../dataMappers/filmDataMapper');
+const userDataMapper = require('../dataMappers/userDataMapper');
+const formatUser = require('../utils/formatUser');
+
+const mainController = {
+
+  renderHomePage(_, res) {
+    res.render('home');
+  },
+
+  redirectHomePage(_, res) {
+    res.redirect('/app');
+  },
+
+  async renderFilmsListPage(_, res) {
+    try {
+      const films = await dataMapper.getdAllFilms();
+      res.render('films', { films });
+    } catch (err) {
+      res.status(500).send('Server Error');
+    }
+  },
+
+  async renderFilmsByReleaseDatePage(req, res) {
+    try {
+      const param = req.params.filmDate;
```

Comment:
```suggestion
      const param = parseInt(req.params.filmDate); 
```

C'est une année qu'on attend ? Si oui, appelle ta variable `filmReleaseYear` ça donne plus d'indication aux devs qui passent après toi